test/unit/webgl/p5.Camera.js

```diff
@@ -0,0 +1,208 @@
+
+    test('Move() back and forth should result in similar eyeXYZ', function() {
+      var originalEyeXYZ = {
+        x: myCam.eyeX,
+        y: myCam.eyeY,
+        z: myCam.eyeZ
+      };
+
+      for (let i = 0; i < 20; i++) {
+        var myVec = {
+          x: Math.random(),
+          y: Math.random(),
+          z: Math.random()
+        };
+        myCam.move(myVec.x, myVec.y, myVec.z);
+        myCam.move(-1 * myVec.x, -1 * myVec.y, -1 * myVec.z);
+      }
+      assert.closeTo(myCam.eyeX, originalEyeXYZ.x, delta);
+      assert.closeTo(myCam.eyeY, originalEyeXYZ.y, delta);
+      assert.closeTo(myCam.eyeZ, originalEyeXYZ.z, delta);
+    });
+
+    test('SetPosition() should set eyeXYZ', function() {
+      var pos = Math.random();
+
+      myCam.setPosition(pos, pos, pos);
+
+      assert.strictEqual(myCam.eyeX, pos);
+      assert.strictEqual(myCam.eyeY, pos);
+      assert.strictEqual(myCam.eyeZ, pos);
+    });
+  });
+
+  suite('Projection', function() {
+    test('ortho() sets renderer uPMatrix', function() {
+      myCam.ortho(-10, 10, -10, 10, 0, 100);
+
+      for (let i = 0; i < myCam.projMatrix.mat4.length; i++) {
+        assert.strictEqual(
+          myCam.projMatrix.mat4[i],
+          myp5._renderer.uPMatrix.mat4[i]
+        );
+      }
+    });
+
+    test('perspective() sets renderer uPMatrix', function() {
+      myCam.perspective(Math.PI / 3.0, 1, 1, 100);
+
+      for (let i = 0; i < myCam.projMatrix.mat4.length; i++) {
+        assert.strictEqual(
+          myCam.projMatrix.mat4[i],
+          myp5._renderer.uPMatrix.mat4[i]
+        );
+      }
+    });
+
+    test('perspective() sets projection matrix correctly', function() {
+      //prettier-ignore
+      var control = [1,  0,   0,  0,
+                     0, -1,   0,  0,
+                     0,  0,  -3, -1,
+                     0,  0, -40,  0];
+
+      myCam.perspective(Math.PI / 2, 1, 10, 20);
+
+      for (let i in myCam.projMatrix.mat4) {
+        assert.strictEqual(myCam.projMatrix.mat4[i], control[i]);
+      }
+    });
+
+    test('ortho() sets projection matrix correctly', function() {
+      //prettier-ignore
+      var control = [1,  0,  0, 0,
+                     0, -1,  0, 0,
+                     0,  0, -1, 0,
+                    -0, -0, -1, 1];
+
+      myCam.ortho(-1, 1, -1, 1, 0, 2);
+
+      for (let i in myCam.projMatrix.mat4) {
+        assert.strictEqual(myCam.projMatrix.mat4[i], control[i]);
+      }
+    });
+  });
+
+  suite('Helper Functions', function() {
+    test('copy() returns a new p5.Camera object', function() {
+      var newCam = myCam.copy();
+      assert.instanceOf(newCam, p5.Camera);
+      assert.notDeepEqual(newCam, myCam);
+    });
+
+    test('_getLocalAxes() returns three vectors', function() {
+      var local = myCam._getLocalAxes();
+      for (let j = 0; j < 3; j++) {
+        assert.typeOf(local.x[j], 'number');
+        assert.typeOf(local.y[j], 'number');
+        assert.typeOf(local.z[j], 'number');
```

Comment:
great. are there any other conditions that should be true here? should the axes be normalized? should they be orthogonal?

Reply:
Yes, those checks are added.